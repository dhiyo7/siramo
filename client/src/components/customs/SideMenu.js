import React, { Component } from 'react'
import {
 Image,
 ImageBackground,
 Text,
 ScrollView,
 View,
 StyleSheet,
 Platform,
 Alert
} from 'react-native';
import {NavigationActions} from 'react-navigation';
import { Ionicons } from '@expo/vector-icons'

class SideMenu extends Component {
  constructor(props) {
    super(props);
    this.state = {  };
  }

  navigateToScreen = (route) => () => {
    const navigateAction = NavigationActions.navigate({
      routeName: route
    });
    this.props.navigation.dispatch(navigateAction);
  }

  logout = () => {
    Alert.alert(
      'WARNING !',
      'Are you sure want to logout?',
      [
        {text: 'Cancel', onPress: () => console.log('Cancel Pressed'), style: 'cancel'},
        {text: 'Logout', onPress: () => this.props.navigation.push('Login')},
      ],
      { cancelable: false }
    )
  }

  render() {
    return (
      <View style={styles.container}>
        <ImageBackground
          style={{width: '100%', height: 160, alignItems: 'center'}}
          source={{uri: 'https://i.ytimg.com/vi/rTVRuEhbVXY/maxresdefault.jpg'}}
        >
        <View style={styles.imageContainer} >
          <Image 
            source={{uri:'https://pbs.twimg.com/profile_images/988775660163252226/XpgonN0X_400x400.jpg'}}
            style={{width: 120, height: 120, borderRadius: 120/2}}
          />
         </View>
        </ImageBackground>
      <ScrollView>
        <View>
          <View style={styles.navSectionStyle}>
            <Text style={styles.navItemStyle} onPress={this.navigateToScreen('Home')}>
              <Ionicons name="md-home" size={24} color="#66BB6A" /> &nbsp;&nbsp; Home
            </Text>
          </View>
        </View>
        <View>
          <View style={styles.navSectionStyle}>
            <Text style={styles.navItemStyle} onPress={this.navigateToScreen('History')}>
              <Ionicons name="md-contacts" size={24} color="#ef5350" /> &nbsp;&nbsp; About
            </Text>
          </View>
        </View>
        <View>
          <View style={styles.navSectionStyle}>
            <Text style={styles.navItemStyle} onPress={this.navigateToScreen('History')}>
              <Ionicons name="md-analytics" size={24} color="#7E57C2" /> &nbsp;&nbsp; History
            </Text>
          </View>
        </View>
        <View>
          <View style={styles.navSectionStyle}>
            <Text style={styles.navItemStyle} onPress={this.logout}>
              <Ionicons name="md-log-out" size={24} color="#ef5350" /> &nbsp;&nbsp; Logout
            </Text>
          </View>
        </View>
      </ScrollView>
      <View style={styles.footerContainer}>
        <Text style={{
          fontSize: 24,
          fontWeight: 'bold',
          color: '#FAFAFA'
        }}>&#169; Siramo</Text>
      </View>
    </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  navItemStyle: {
    padding: 10,
    fontSize: 20,
  },
  navSectionStyle: {
  },
  sectionHeadingStyle: {
    paddingVertical: 10,
    paddingHorizontal: 5
  },
  footerContainer: {
    backgroundColor: '#A1887F',
    alignItems: 'center',
    justifyContent: 'center'
  },
  imageContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: ( Platform.OS === 'ios' ) ? 20 : 0
  }
})
export default SideMenu;