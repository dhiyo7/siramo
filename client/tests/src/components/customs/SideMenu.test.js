import React from 'react'
import {
  Image,
  ImageBackground,
  Text,
  ScrollView,
  View,
  StyleSheet,
  Platform,
  Alert,
} from 'react-native'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import SideMenu from '../../../../src/components/customs/SideMenu'
import { Ionicons } from '@expo/vector-icons'

Enzyme.configure({ adapter: new Adapter()})

let wrapper

beforeEach(() => {
  const navigation = { navigate: jest.fn(), dispatch: jest.fn()}
  wrapper = shallow(<SideMenu navigation={navigation} />)
})

describe('<SideMenu /> snapshot Testing', () => {
  it('snapshot works!', () => {
    const tree = renderer.create(<SideMenu />)
    expect(tree).toMatchSnapshot()
  })
})

describe('<SideMenu />', () => {
  it('should render <SideMenu />', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('<SideMenu /> child component Test', () => {
  it('<SideMenu /> should render all children Component', () => {
    Alert.alert = jest.genMockFunction()
    expect(wrapper.find('ImageBackground').get(0).props.source.uri).toEqual('https://i.ytimg.com/vi/rTVRuEhbVXY/maxresdefault.jpg')
    expect(wrapper.find('Image').get(0).props.source.uri).toEqual('https://pbs.twimg.com/profile_images/988775660163252226/XpgonN0X_400x400.jpg')
    expect(wrapper.find('Text').length).toBe(5)
    expect(wrapper.find('Text').get(0).props.children[0]).toEqual(<Ionicons name="md-home" size={24} color="#66BB6A" />)
    // expect(wrapper.find('Text').get(0).props.children[1]).toBe('    Home')
    expect(wrapper.find('Text').get(1).props.children[0]).toEqual(<Ionicons name="md-calendar" size={24} color="#ef5350" />)
    // expect(wrapper.find('Text').get(1).props.children[1]).toBe('    Scheduling')
    expect(wrapper.find('Text').get(2).props.children[0]).toEqual(<Ionicons name="md-analytics" size={24} color="#7E57C2" />)
    // expect(wrapper.find('Text').get(2).props.children[1]).toBe('    History')
    expect(wrapper.find('Text').get(3).props.children[0]).toEqual(<Ionicons name="md-log-out" size={24} color="#ef5350" />)
    wrapper.find('Text').get(3).props.onPress()
    // console.log(wrapper.get(0))
    // expect(wrapper.find('Text').get(3).props.children[1]).toBe('    Logout')
    expect(wrapper.find('Text').get(4).props.style).toEqual({ fontSize: 24, fontWeight: 'bold', color: '#FAFAFA' })
    expect(wrapper.find('Text').get(4).props.children).toEqual('Â© Siramo')
  })
})