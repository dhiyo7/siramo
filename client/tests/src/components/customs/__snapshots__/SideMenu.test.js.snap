// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideMenu /> snapshot Testing snapshot works! 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 160,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://i.ytimg.com/vi/rTVRuEhbVXY/maxresdefault.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 160,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingTop": 20,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://pbs.twimg.com/profile_images/988775660163252226/XpgonN0X_400x400.jpg",
          }
        }
        style={
          Object {
            "borderRadius": 60,
            "height": 120,
            "width": 120,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View>
        <View
          style={Object {}}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={[Function]}
            style={
              Object {
                "fontFamily": "sans-serif-condensed",
                "fontSize": 20,
                "padding": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
                Home
          </Text>
        </View>
      </View>
      <View>
        <View
          style={Object {}}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={[Function]}
            style={
              Object {
                "fontFamily": "sans-serif-condensed",
                "fontSize": 20,
                "padding": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
                Scheduling
          </Text>
        </View>
      </View>
      <View>
        <View
          style={Object {}}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={[Function]}
            style={
              Object {
                "fontFamily": "sans-serif-condensed",
                "fontSize": 20,
                "padding": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
                History
          </Text>
        </View>
      </View>
      <View>
        <View
          style={Object {}}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={[Function]}
            style={
              Object {
                "fontFamily": "sans-serif-condensed",
                "fontSize": 20,
                "padding": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
                Logout
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#A1887F",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FAFAFA",
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      © Siramo
    </Text>
  </View>
</View>
`;
